src/ccsr/src/client.rs

```diff
@@ -83,6 +83,23 @@ impl Client {
         Ok(res.id)
     }
 
+    /// Same as `publish_schema`, but also includes a list of `SchemaReference`s.
+    pub async fn publish_schema_with_references(
```

Comment:
I made this a separate function because we only need to use it for a test, but happy to refactor the main function as well!

Reply:
Yeah, I think just make the `references` parameter part of the main function, and pass in `&[]` as necessary!